blaze-server/src/main/scala/org/http4s/blaze/server/Http1ServerStage.scala

```diff
@@ -303,17 +302,21 @@ private[blaze] class Http1ServerStage[F[_]](
             logger.trace("Request/route requested closing connection.")
             F.delay(closeConnection())
           } else
-            F.delay {
-              bodyCleanup().onComplete {
-                case s @ Success(_) => // Serve another request
-                  parser.reset()
-                  handleReqRead(s)
-
-                case Failure(EOF) => closeConnection()
-
-                case Failure(t) => fatalError(t, "Failure in body cleanup")
-              }(trampoline)
-            }
+            F.async_[Unit] { cb =>
+              cb(
+                Right(
+                  bodyCleanup().onComplete {
+                    case s @ Success(_) => // Serve another request
+                      parser.reset()
+                      handleReqRead(s)
+
+                    case Failure(EOF) => closeConnection()
+
+                    case Failure(t) => fatalError(t, "Failure in body cleanup")
+                  }(trampoline)
+                )
+              )
+            } >> F.cede
```

Comment:
Probably `async` will be enough (because of auto-ceding) and this `cede` is redundant. This needs CE's expert opinion. I added it just because we have to deal with external EC.

Reply:
Yes, I thought it's a bit odd. When an async completes it reschedules the fiber onto pool anyway. Adding a `cede` so soon will de-schedule it again.